File: templates/java/util/FileUtil.java

import java.util.*;
import java.io.*;

public class FileUtil {
    
    private static final String DEFAULT_CHARSET = "utf-8";
    /**
     * Write content into file, default use utf-8 encode
     * @param filename
     * @param content
     */
    public static void writeFile(String filename, String content) {
        writeFile(filename, content, DEFAULT_CHARSET);
    }
    /**
     * Write content into file
     * @param filename
     * @param content
     * @param charset
     */
    public static void writeFile(String filename, String content, String charset) {
        File file = new File(filename);
        try(OutputStreamWriter osw = new OutputStreamWriter(new FileOutputStream(file), charset);
        BufferedWriter bw = new BufferedWriter(osw);

        ) {
            bw.write(content);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
    /**
     * Append content into file, default use utf-8 encode
     * @param filename
     * @param content
     */
    public static void appendFile(String filename, String content) {
        appendFile(filename, content, DEFAULT_CHARSET);
    }
    /**
     * Append content into file
     * @param filename
     * @param content
     * @param charset
     */
    public static void appendFile(String filename, String content, String charset) {
        File file = new File(filename);
        try(OutputStreamWriter osw = new OutputStreamWriter(new FileOutputStream(file), charset);
        BufferedWriter bw = new BufferedWriter(osw);) {
            bw.append(content);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    /**
     * Read file content into a string list.
     * @param filename
     * @return
     */
    public static List<String> readLines(String filename) {
        return readLines(filename, DEFAULT_CHARSET);
    }
    public static List<String> readLines(String filename, String charset) {
        File file = new File(filename);
        ArrayList<String> result = new ArrayList<>();
        if (!file.exists()) {
            return result;
        }
        try(InputStreamReader fr = new InputStreamReader(new FileInputStream(file), charset);) {
            
            BufferedReader br = new BufferedReader(fr);
            String line = br.readLine();
            while(line != null){
                result.add(line);
                line = br.readLine();
            }
            
        } catch (IOException e) {
            e.printStackTrace();
        }

        return result;
    }

    /**
     * Read file content as string
     * @param filename
     * @return
     */
    public static String readFile(String filename) {
        return readFile(filename, DEFAULT_CHARSET);
    }
    public static String readFile(String filename, String charset) {
        File file = new File(filename);
        String result = "";
        if (!file.exists()) {
            return result;
        }
        StringBuilder sb = new StringBuilder();
        try(InputStreamReader fr = new InputStreamReader(new FileInputStream(file), charset)) {
            int readed = fr.read();
            while(readed != -1){

                sb.append((char)readed);
                readed = fr.read();
            }
            result = sb.toString();
            
        } catch (IOException e) {
            e.printStackTrace();
        }

        return result;
    }
}

File: templates/python/util/file_util.py
# -*- coding: utf-8 -*-


# 列表内容写入文件。覆盖
def write_list(filename, list, toString=None):
    content = getListContent(list, toString)
    writeFile(filename, content)

# 列表内容转换为字符串
def get_list_content(list, toString=None):
    content = ""
    for item in list:
        content = content + (toString(item) if toString != None else str(item)) + "\n"
    return content
# 列表内容写入文件。 追加 
def append_list(filename, list, toString=None):
    content = getListContent(list, toString)
    appendFile(filename, content)

# 写入文件内容。覆盖
def write_file(filename, content):
    file_object = open(filename, 'w')
    file_object.write(content)
    file_object.close()

# 追加文件内容
def append_file(filename, content):
    file_object = open(filename, 'a')
    file_object.write(content)
    file_object.close()

# 读取文件内容
def read_file(filename):
    with open(filename, 'r') as file:
        content = file.read()
    return content

def read_lines(filename):
    file = open(filename)
    lines = [line.strip('\n') for line in file.readlines()]
    file.close()
    return lines
